CHANGELOG.md

```diff
@@ -1,5 +1,40 @@
 # Changelog
 
+## 1.4.0
+
+Features/improvements:
+
+- Correct processing of modifications and deletions.  In prior versions, Pachyderm pipelines can only process data additions; data that are removed or modified are effectively ignored.  In 1.4, when certain input data are removed (or modified), downstream pipelines know to remove (or modify) the output that were produced as a result of processing the said input data.
+
+As a consequence of this change, a user can now fix a pipeline that has processed erroneous data by simply making a new commit that fixes the said erroneous data, as opposed to having to create a new pipeline.
+
+- Vastly improved performance for metadata operations (e.g. list-file, inspect-file).  In prior versions, metadata operations on commits that are N levels deep are O(N) in runtime.  In 1.4, metadata operations are always O(1), regardless of the depth of the commit. 
+
+- A new way to specify how input data is partitioned.  Instead of using two flags `partition` and `incrementality`, we now use a single `glob` pattern.  See the glob doc for details.
+
+- Flexible branch management.  In prior versions, branches are fixed, in that a commit always stays on the same branch, and a branch always refers to the same series of commits.  In 1.4, branches are modeled similar to Git's tags; they can be created, deleted, and renamed indepedently of commits.
+
+- Simplified commit states.  In prior versions, commits can be in many states including `open`, `closed`, `cancelled`, and `archived`.  In particular, `closed` and `archived` have confusing semantics that routinely trip up users.  In 1.4, `closed` and `archived` have been removed.
```

Comment:
I think you mean `cancelled` and `archived` where removed? not `closed`. Are open and closed actually the status's we use or is it started and finished? Either is fine for this doc I think.